tests/models/test_lisflood.py

```diff
@@ -109,3 +112,61 @@ def test_setup(self, model_with_setup, tmp_path):
             textvar_name = textvar.attrib["name"]
             if textvar_name == 'IrrigationEfficiency':
                 assert textvar.get('value') == '0.8'
+
+    class TestGetValueAtCoords():
+
+        def test_get_value_at_coords_single(self, model: Lisflood):
+            expected = np.array([1.0])
+            model.bmi = MockedBmi()
+            actual = model.get_value_at_coords('Discharge', lon=[-124.35], lat=[52.93])
+            assert_array_equal(actual, expected)
+            assert model.bmi.indices == [311]
+
+        def test_get_value_at_coords_multiple(self, model: Lisflood):
+            model.bmi = MockedBmi()
+            model.get_value_at_coords('Discharge', lon=[-124.45, -124.35, -121.45], lat=[53.95, 52.93, 52.65])
+            assert_array_equal(model.bmi.indices, [0, 311, 433])
+
+        def test_get_value_at_coords_faraway(self, model: Lisflood):
+            model.bmi = MockedBmi()
+            with pytest.raises(ValueError) as excinfo:
+                    model.get_value_at_coords('Discharge', lon=[0.0], lat=[0.0])
+            msg = str(excinfo.value)
+            assert "This point is outside of the model grid." in msg
+
+
+class MockedBmi(Bmi):
+    def get_var_grid(self, name):
+        return 0
+
+    def get_grid_shape(self, grid_id):
+        return (14, 31) #shape returns (len(y), len(x))
+
+    def get_grid_x(self, grid_id):
+         return np.array([-124.450000000003, -124.350000000003, -124.250000000003,
+                -124.150000000003, -124.050000000003, -123.950000000003,
+                -123.850000000003, -123.750000000003, -123.650000000003,
```

Comment:
2 questions here:

- Why use this strange high precision ending at 000003 or 00000021?
- Since it's a mock, why not make a much smaller grid of, say, `2x3`?

Reply:
> 2 questions here:
> 
> * Why use this strange high precision ending at 000003 or 00000021?

This format is actually returned by bmi too. This was the reason where I added a` round( , 4)` in the code.  User does not see the high precision. 

> * Since it's a mock, why not make a much smaller grid of, say, `2x3`?

Agree, but the test is using a real sample data, see class `TestLFlatlonUseCase` in the test.